CommandLine/libEGM/action.cpp

```diff
@@ -113,13 +109,45 @@ std::string Actions2Code(const std::vector< buffers::resources::Action >& action
       case ActionKind::ACT_CODE:
         code += "{\n" + args.Get(0).string() + "\n/**/\n}";
         break;
-      case ActionKind::ACT_NORMAL:
+      case ActionKind::ACT_NORMAL: {
         if (action.exe_type() == ActionExecution::EXEC_NONE) break;
 
-        code += action.relative() ? "argument_relative = true;\n" : "argument_relative = false;\n";
+        const std::string &applies_to = action.who_name();
+
+        const auto &ref_as_int = [&action](const std::string &applies_to) -> std::string {
+          if (applies_to == "self") {
+            return "-1";
+          } else if (applies_to == "other") {
+            return "-2";
+          } else if (action.id() == -4) {
+            return "-100";
+          } else {
+            return std::to_string(action.id());
+          }
+        };
+
+        if (action.use_apply_to() && applies_to != "self") {
+          if (action.is_question()) {
+            if (applies_to == "other") {
+              code += "with (other) ";
+            } else if (applies_to != "") {
+              code += std::string{"with ("} + ref_as_int(applies_to) + ") ";
+            } else {
+              code += "/*null with*/";
+            }
+          } else {
+            if (applies_to == "other") {
+              code += "with (other) {";
+            } else if (applies_to != "") {
+              code += std::string{"with ("} + ref_as_int(applies_to) + ") {";
+            } else {
+              code += "/*null with*/{";
+            }
+          }
+        }
 
         if (action.is_question()) {
-          code += "__if__ = ";

```

Comment:
As discussed, you can't remove this fix that I added for if statement scoping.
Otherwise you'll break the scoping of variables inside if actions which can also have applies to.
The applies to/calling if inside a with statement is necessary so we don't have to prepend `self.` to all the variables in the if expression.

Reply:
As per your suggestions, I have reverted the changes and added `__if__` back.